stl/src/stacktrace.cpp

```diff
@@ -0,0 +1,344 @@
+                                                    | 0x4 /* SYMOPT_DEFERRED_LOADS */
+                                                    | 0x10 /* SYMOPT_LOAD_LINES */
+                                                    | 0x20 /* SYMOPT_OMAP_FIND_NEAREST */
+                                                    | 0x100 /* SYMOPT_FAIL_CRITICAL_ERRORS */
+                                                    | 0x10000 /* SYMOPT_AUTO_PUBLICS */
+                                                    | 0x80000 /* SYMOPT_NO_PROMPTS */;
+
+                        constexpr ULONG remove_options = 0x8 /* SYMOPT_NO_CPP */
+                                                       | 0x40 /* SYMOPT_LOAD_ANYTHING */
+                                                       | 0x100 /* SYMOPT_NO_UNQUALIFIED_LOADS */
+                                                       | 0x400 /* SYMOPT_EXACT_SYMBOLS */
+                                                       | 0x1000 /* SYMOPT_IGNORE_NT_SYMPATH */
+                                                       | 0x4000 /* SYMOPT_PUBLICS_ONLY */
+                                                       | 0x8000 /* SYMOPT_NO_PUBLICS */
+                                                       | 0x20000 /* SYMOPT_NO_IMAGE_SEARCH */;
+
+                        (void) debug_symbols->AddSymbolOptions(add_options);
+                        (void) debug_symbols->RemoveSymbolOptions(remove_options);
+                    }
+                }
+            }
+
+            return attached;
+        }
+
+        size_t get_description(
+            const void* const address, void* const str, size_t off, const _Stacktrace_string_fill fill) {
+            // Initially pass the current capacity, will retry with bigger buffer if it fails.
+            size_t size          = fill(0, str, nullptr, nullptr) - off;
+            HRESULT hr           = E_UNEXPECTED;
+            ULONG64 displacement = 0;
+
+            for (;;) {
+                ULONG new_size = 0;
+
+                const size_t new_off = string_fill(
+                    fill, off + size, str, [address, off, size, &new_size, &hr, &displacement](char* s, size_t) {
+                        hr = debug_symbols->GetNameByOffset(reinterpret_cast<uintptr_t>(address), s + off,
+                            static_cast<ULONG>(size + 1), &new_size, &displacement);
+
+                        return (hr == S_OK) ? off + new_size - 1 : off;
+                    });
+
+                if (hr == S_OK) {
+                    off = new_off;
+                    break;
+                } else if (hr == S_FALSE) {
+                    size = new_size - 1; // retry with bigger buffer
+                } else {
+                    return off;
+                }
+            }
+
+            if (displacement != 0) {
+                constexpr size_t max_disp_num = sizeof("+0x1111222233334444") - 1; // maximum possible offset

```

Comment:
```suggestion
                constexpr size_t max_disp_num = sizeof("+0x1122334455667788") - 1; // maximum possible offset
```

this obviously doesn't make a difference in code, but it makes obvious the byte boundaries.

Reply:
Also, should this be `sizeof("+0x11223344")` on 32-bit platforms?

Reply:
`sizeof(ULONG64) == 8`

In practice, PE format does not allow that much of code even for 64 bits, so there would be at most 8 hexits even in 64-bit, but I don't want to rely on such implicit limits, so used the type limit

Reply:
ah, didn't realize they used `ULONG64` instead of `size_t`